Skip backport packages for newer python

orlitzky opened the pull request:
Three of our SPKGs are backports that provide python-3.11 features:

* importlib_metadata
* importlib_resources
* typing_extensions

As a result, they are not needed when sage's python is v3.11 or newer. We un-require them in `spkg-configure.m4`

orlitzky committed:
build/pkgs/importlib_metadata/install-requires.txt: skip if python >= 3.11

orlitzky committed:
build/pkgs/importlib_resources/install-requires.txt: skip if python >= 3.11

orlitzky committed:
build/pkgs/typing_extensions/install-requires.txt: skip if python >= 3.11

mkoeppe commented on build/pkgs/importlib_metadata/spkg-configure.m4:
```diff
@@ -1,3 +1,26 @@
+  # These packages are therefore not needed with >=python-3.11. Here
+  # we test for a python minor version component greater than or equal
+  # to 11, and mark this package as "not required" if we succeed.
+  AC_MSG_CHECKING([for >=python-3.11])
```
Why does the Python version need a separate check in `spkg-configure.m4`? 
Does the conditionalization in `install-requires.txt` not do the job already? Checking whether `importlib_metadata >=4.13; python_version<'3.11'` is available will be trivially true for Python >= 3.11

orlitzky replied:
Oh. I wrote that check before you pointed out that we needed to change install-requires.txt. It's probably redundant now but I have to get flint-3.x packaged before I can run `./configure` now.

orlitzky replied:
I had to change the quoting in the macro but it basically works, if you don't care that it says "will use the system foo" rather than "foo is not required" when it skips the package.

mkoeppe replied:
I think that's fine

orlitzky committed:
m4/sage_python_package_check.m4: double-quote the version spec

Some version specifiers in install-requires.txt will contain single
quotes; for example, python_version<'3.11'. (This is according to the
setuptools "Platform Specific Dependencies" documentation.) To support
that, we double-quote the spec string rather than single quote it.

mkoeppe commented on m4/sage_python_package_check.m4:
```diff
@@ -75,10 +75,13 @@ AC_DEFUN([SAGE_PYTHON_PACKAGE_CHECK], [
 
+      dnl double-quote SAGE_PKG_VERSPEC because platform-specific
+      dnl dependencies like python_version<'3.11' will have single
+      dnl quotes in them.
```
Actually, you can use double quotes in install-requires.txt.

And in fact double quotes are better because then `m4/sage_spkg_versions_toml.m4` will be generated correctly!

mkoeppe replied:
(relevant for #36951)

orlitzky replied:
Ok, double quotes it is. I changed the package-check macro to normalize them (back to single quotes) so that both are supported, at least in that one macro.

orlitzky committed:
build/pkgs: double-quote platform dependencies in install-requires.txt

For the moment, this is an easy fix for the ./bootstrap generation
of sage_spkg_versions_toml.m4, which single-quotes the contents
of install-requires.

orlitzky committed:
m4/sage_python_package_check.m4: normalize constraint quotes

Platform version constraints in install-requires can be either single-
or double-quoted. To keep things simple, we'd like to assume in the
macro that they are single-quoted, following the examples in the
documentation. We now use sed to normalize them.

mkoeppe reviewed (approved):
LGTM, thanks.
